ui/core/resources/merger.ts

```diff
@@ -0,0 +1,23 @@
+import { ResourceType } from '../proto/api';
+import { ResourceChangedLogGroup } from '../proto_utils/logs_parser';
+
+export class ResourceMerger {
+	protected readonly targets: ResourceType[];
+	protected readonly aggregate_key: ResourceType;
+
+	public constructor(targets: ResourceType[], aggregate_key: ResourceType) {
+		this.targets = targets;
+		this.aggregate_key = aggregate_key;
+	}
+
+	public mergeResources(resourceEvents: Record<ResourceType, ResourceChangedLogGroup[]>): Record<ResourceType, ResourceChangedLogGroup[]> {
```

Comment:
Seems like this should just be a static function with 3 parameters (adding `targets` and `aggregate_key`? This object is never used beyond a single call to the function.

Reply:
Actually, another alternative if you want to stick with the OO pattern: add a static `RUNE_MERGER` instance of this class which only gets created once.

Reply:
The idea behind making this a generic class is that you could do something like:

```ts
const health_mana_merger = new new ResourceMerger([ResourceType.ResourceTypeHealth, ResourceType.ResourceTypeMana], ResourceType.ResourceTypeHealth)
health_mana_merger.mergeResources(some_input)
```

Which enables you to do this for any set of resources, not only runes.